list.js

```diff
@@ -3,6 +3,7 @@
 module.exports = List;
 
 var Shim = require("./shim");
+var Iterator = require("./iterator");
```

Comment:
I've found that the code containing require('collection/list') and calling list.iterator() generates unpredictable and mess error in the code of list implementation, if the code using list doesn't contain require('collection/iterator'). This line fixes this condition.